Assets/Shopify/Unity/CartState.cs

```diff
@@ -43,11 +43,12 @@ public partial class CartState {
 
         public CartState(ShopifyClient client) {
             Client = client;
-            _LineItems = new CartLineItems(OnDeleteLineItem);
+            _LineItems = new CartLineItems();
+            _LineItems.OnChange += OnLineItemChange;
```

Comment:
Same thing here about this potential getting re-registered.

Reply:
You can't really re-register here since the `CartLineItems` instance is created in the constructor for `CartState`. But you are right that we potentially need to have a destructor/cleanup function.